src/landIce/problems/LandIce_LaplacianSampling.hpp

```diff
@@ -145,29 +145,63 @@ LandIce::LaplacianSampling::constructEvaluators (PHX::FieldManager<PHAL::AlbanyT
   int total_num_param_vecs, num_param_vecs, num_dist_param_vecs;
   Albany::getParameterSizes(parameterParams, total_num_param_vecs, num_param_vecs, num_dist_param_vecs);
 
-  stateName = "weighted_normal_sample";
-  bool isParameter = false;
-  for (unsigned int p_index=0; p_index< (unsigned int) num_dist_param_vecs; ++p_index) {
-    std::string parameter_sublist_name = util::strint("Parameter", p_index+num_param_vecs);
-    Teuchos::ParameterList param_list = parameterParams.sublist(parameter_sublist_name);
-    param_name = param_list.get<std::string>("Name");
-    if(param_name == stateName) {
-      isParameter = true;
-      break;
+  {
+    stateName = "weighted_normal_sample";
+    bool isParameter = false;
+    for (unsigned int p_index=0; p_index< (unsigned int) num_dist_param_vecs; ++p_index) {
```

Comment:
Using `int` rather than `unsigned int` for `p_index` would allow to avoid the cast, and keep the line more readable...
```suggestion
    for (int p_index=0; p_index<num_dist_param_vecs; ++p_index) {
```

Reply:
I know you didn't change this, but while at it...